spec/tunes/app_version_spec.rb

```diff
@@ -150,10 +171,308 @@
+          r = [900, 1200] if path == ipad_trailer_path
+          r = [768, 1024] if path == ipad_trailer_preview_path
+          r = [700, 1000] if path == ipad_external_invalid_trailer_preview_path
+          r = [768, 1024] if path == ipad_external_valid_trailer_preview_path
+          r
+        end
+
+        json = JSON.parse(du_read_fixture_file("upload_trailer_response_success.json"))
+        allow(client.du_client).to receive(:upload_trailer).and_return(json)
+
+        json = JSON.parse(du_read_upload_trailer_preview_response_success)
+        allow(client.du_client).to receive(:upload_trailer_preview).and_return(json)
+      end
+
+      def trailers(device)
+        version.trailers["English"].find_all { |s| s.device_type == device }
+      end
+
+      def ipad_trailers
+        trailers("ipad")
+      end
+
+      it "cannot add a trailer to iphone35" do
+        expect do
+          version.upload_trailer!(ipad_trailer_path, "English", 'iphone35')
+        end.to raise_error "No app trailer supported for iphone35"
+      end
+
+      it "requires timestamp with a specific format" do
+        expect do
+          version.upload_trailer!(ipad_trailer_path, "English", 'ipad', "00:01.000")
+        end.to raise_error "Invalid timestamp 00:01.000"
+        expect do
+          version.upload_trailer!(ipad_trailer_path, "English", 'ipad', "01.000")
+        end.to raise_error "Invalid timestamp 01.000"
+      end
+
+      it "can add a new trailer" do
+        # remove existing
+        version.upload_trailer!(nil, "English", 'ipad')
+
+        count = ipad_trailers.count
+        expect(count).to eq(0)
+        version.upload_trailer!(ipad_trailer_path, "English", 'ipad')
+        count_after = ipad_trailers.count
+        expect(count_after).to eq(count + 1)
+        expect(count_after).to eq(count + 1)
+        trailer = ipad_trailers[0]
+        expect(trailer.video_asset_token).to eq("VideoSource40/v4/e3/48/1a/e3481a8f-ec25-e19f-5048-270d7acaf89a/pr_source.mov")
+        expect(trailer.picture_asset_token).to eq("Purple69/v4/5f/2b/81/5f2b814d-1083-5509-61fb-c0845f7a9374/pr_source.jpg")
+        expect(trailer.descriptionXML).to match(/FoghornLeghorn/)
+        expect(trailer.preview_frame_time_code).to eq("00:00:05:00")
+        expect(trailer.video_url).to eq(nil)
+        expect(trailer.preview_image_url).to eq(nil)
+        expect(trailer.full_sized_preview_image_url).to eq(nil)
+        expect(trailer.device_type).to eq("ipad")
+        expect(trailer.language).to eq("English")
+      end
+
+      it "can modify the preview of an existing trailer and automatically generates a new screenshot preview" do
+        json = JSON.parse(du_read_upload_trailer_preview_2_response_success)
+        allow(client.du_client).to receive(:upload_trailer_preview).and_return(json)
+
+        count = ipad_trailers.count
+        expect(count).to eq(1)
+        version.upload_trailer!(ipad_trailer_path, "English", 'ipad', "06.12")
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.